alns/stopping_criteria/MaxIterations.py

```diff
@@ -0,0 +1,27 @@
+from alns.State import State
+from alns.stopping_criteria.StoppingCriterion import StoppingCriterion
+
+
+class MaxIterations(StoppingCriterion):
+    def __init__(self, max_iterations: int) -> None:
+        """
+        Criterion that stops after a maximum number of iterations.
+        """
+        if max_iterations < 0:
+            raise ValueError("Max iterations must be non-negative.")
+
+        self._max_iterations = max_iterations
+        self._current_iteration = 0
+
+    @property
+    def max_iterations(self) -> int:
+        return self._max_iterations
+
+    @property
+    def current_iteration(self) -> int:
```

Comment:
Refactor: I am not sure what exposing this field adds to the programme. I suggest we keep `_current_iterations` internal/private, so not have this property.

Reply:
That would be roughly in line with the acceptance criteria, which also do not expose threshold and temperatures. Only the initial parameters.

Reply:
See [15540a0](https://github.com/N-Wouda/ALNS/pull/64/commits/15540a079f9b1aaddbb9a4e03aa15c6ccb18b57b)